server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -783,19 +783,23 @@ private void findStorageGroup(MNode node, String path, HashSet<String> res) {
    * @return a list contains all distinct devices
    */
   Set<String> getAllDevices() {
-    return new HashSet<>(getNodesList(3));
+    return new HashSet<>(getNodesList("root",2));
   }
 
   /**
    * Get all nodes at the given level in current Metadata Tree.
    *
    * @return a list contains all nodes at the given level
    */
-  List<String> getNodesList(int nodeLevel) {
+  List<String> getNodesList(String schemaPattern, int nodeLevel) {
     List<String> res = new ArrayList<>();
-    MNode rootNode;
-    if ((rootNode = getRoot()) != null) {
-      findNodes(rootNode, "root", res, nodeLevel);
+    String[] nodes = schemaPattern.split("\\.");
+    MNode node;
+    if ((node = getRoot()) != null) {
+      for (int i = 1; i < nodes.length; i++) {
+        node = node.getChild(nodes[i]);
```

Comment:
What if the node does not have such a child.

Reply:
@LeiRui Hi,I have no idea about how to implement this as the function of showing all devices was not developed by me. I just did some minor changes to ensure that it won't report the error.

Could you please provide me some advices or discuss with me on how to get this function done? Maybe this should be on another PR. What do you guys think.